fitbenchmarking/fitbenchmarking
# Add `validate` method to base controller to check compatibility for `cs` jacobian

#### Description of Work
This PR adds a validate method to the base controller so that the compatibility of certain options and problem sets can be checked before running a benchmark combination.

This would be a good place to add future compatibility checks that might be needed in the future.

There is a `validate_minimizer` function in the base controller that could probably be made private and moved to the same place as the `_validate_jacobian` function call. The validation in the `setup` function could also be moved here in the future.

Fixes #535

#### Testing Instructions

1. Use the following options file
```
[MINIMIZERS]
scipy_ls: lm-scipy

[JACOBIAN]
scipy: cs

[FITTING]
num_runs: 1
software: scipy_ls
jac_method: scipy
```
with `fitbenchmarking -o examples/options_template.ini -p examples/benchmark_problems/simple_tests/`
2. Check that the messages output when running the mantid problem sets are clear.

Function: Does the change do what it's supposed to?

Tests: Does it pass? Is there adequate coverage for new code?

Style: Is the coding style consistent? Is anything overly confusing?

Documentation: Is there a suitable change to documentation for this change?
